include/class.forms.php

```diff
@@ -2415,15 +2415,17 @@ function getSearchQ($method, $value, $name=false) {
         case 'before':
             return new Q(array("{$name}__lt" => $value));
         case 'between':
-            foreach (array('left', 'right') as $side) {
-                $value[$side] = is_int($value[$side])
-                    ? DateTime::createFromFormat('U', !$config['gmt']
-                        ? Misc::gmtime($value[$side]) : $value[$side]) ?: $value[$side]
-                    : $value[$side];
-            }
+            $tz = new DateTimeZone($cfg->getTimezone());
+            $dbtz = new DateTimeZone($cfg->getDbTimezone());
+            $left = Format::parseDateTime($value['left']);
+            $right = Format::parseDateTime($value['right']);
+            $left = $left->setTime(00, 00, 00);
+            $right = $right->setTime(23, 59, 59);
+            $left->setTimezone($tz)->setTimezone($dbtz);
+            $right->setTimezone($tz)->setTimezone($dbtz);
```

Comment:
I'm not so sure we should convert time to user's timezone. 

I added it because - between widget doesn't provide for opportunity specify timezone but I'll be interested in y'alls feedback.

Reply:
I think it's a good idea as the Date Picker is shown in the user's timezone. So if they chose between `11/11/2019` and `11/12/2019` in the date picker they would expect the system to show results in that exact timeframe in their timezone. I tested this pull and it works as expected. It pulls the correct data.

Reply:
The issue is mute when it comes to the agent doing Advanced Search - the date picture already registers the user's timezone as part of the entry. So we don't actually need to convert to user's timezone in this case.

My concern is when `Between` criteria is used in a global queue. The saved values will be on the queue creator's timezone - forcing conversion to the agent timezone might not be the intended results -  I added it anyhow because `Period` floats the timezone, but it's not specific tho.